### PR TITLE
Hypershift: Integrate new create cluster catalog to current wizard

### PR DESCRIPTION
**Pull request by fxiang1**
Issue:
https://github.com/stolostron/backlog/issues/24372

- Hook up new catalog to wizard
- Moved AI warning to cluster details in wizard

### CODE REVIEW

File: frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateInfrastructure/CreateDiscoverHost.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    const onBack = useCallback(() => history.push(NavigationPath.createInfrastructure), [history])
+
+    return (
+        <Fragment>
+            <PageHeader
+                title={t('Hosts')}
+                description={t('Choose an option based on your hosts.')}
+                breadcrumbs={breadcrumbs}
+            />
+            <ItemView items={cards} itemKeyFn={keyFn} itemToCardFn={(card) => card} onBack={onBack} />
```
[the remark this answers is not in the document]
Comment (fxiang1) [reply frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateInfrastructure/CreateDiscoverHost.tsx]: Yep cancel button added.

File: frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateInfrastructure/CreateDiscoverHost.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        const newBreadcrumbs: ICatalogBreadcrumb[] = [
+            { label: t('Clusters'), to: NavigationPath.clusters },
+            { label: t('Infrastructure'), to: NavigationPath.createInfrastructure },
+            { label: t('Control Plane'), to: NavigationPath.createControlPlane },
+            { label: t('Hosts') },
+        ]
+        return newBreadcrumbs
+    }, [t])
+
+    const onBack = useCallback(() => history.push(NavigationPath.createInfrastructure), [history])
```
[the remark this answers is not in the document]
Comment (fxiang1) [reply frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateInfrastructure/CreateDiscoverHost.tsx]: Memoize removed. Thanks!

**Comment by fxiang1** [comment]
Not sure why the code smell is based on the older versions of the changed files. Looking at the comparisons, the code that its complaining about is not there anymore.

**Review by chenz4027 — APPROVED** [review]
lgtm, thanks Feng!

**Pull request merged**